### DISCUSSION TITLE
How to read the coordinates of a grid in the ProblemGenerator function?

### DISCUSSION THREAD

**Discussion by tianninglyu** [PrincetonUniversity/athena#495] [Q&A] [answered]
Hello everyone, I am new in athena++ and now studying it.
I want to simulate a jet and I start with the problem generator jet.cpp. In order to define an inhomogeneous medium, I define a density function Real Density(Real x, Real y, Real z).
In the function void MeshBlock::ProblemGenerator(Parameter *pin), I want to call the function Density to initialize each grid so that I need to use pco to gain the coordinates of the grid, for example, x1 = pco->x1v(i); . However, pco is defined in the scope "Mesh", not in "MeshBlock", so I can not use pco in the ProblemGenerator function. 
So briefly, my question is how to read the coordinates of a grid in the ProblemGenerator function.
Thank you!

**Comment by c-white** [comment, marked as answer]
The coordinates are located on the MeshBlock: `pcoord->x1v(i)` will work inside `ProblemGenerator()`.

In the jet problem generator, `pco` is just the name given to the input to the boundary condition function. This could be called anything, but it doesn't actually live on the Mesh.

**Reply by tianninglyu** [reply]
Oh, god! I am so sorry that I have made such a funny mistake! Thank you for your consideration!
Actually I wrote a correct code before, using pcoord correctly. I define a cylinar-symmetric density distribution proportional to r^alpha. So, when I set alpha to 0, the result should be the same as in the case of homogeneous medium, but this is not the case. So I started trying to modify my code, and I made such a mistake during the revision process. 
Sincere gratitude to you!
